processors/hra/extractors.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from .. import base
+
+
+# Module API
+
+def extract_source(record):
+    source = {
+        'id': 'hra',
+        'name': 'Health Research Authority'
+    }
+    return source
+
+
+def extract_trial(record):
+
+    # Get identifiers
+    identifiers = base.helpers.clean_dict({
+        'hra_internal_identifier': record['application_id'],
+        'iras_project_identifier': record['iras_proj_id'],
+        'study_type_id': record['study_type_id']
```

Comment:
Yes we need here ids in canonical form `NCT11111*`, `ISRCTN123*`, `EUCTR2011-003574-84-GB`. Ids we can't make compliant we don't need to add to `identifiers` list.